packages/manager/apps/web/client/app/email-domain/upgrade/upgrade.controller.js

```diff
@@ -0,0 +1,89 @@
+import head from 'lodash/head';
+import get from 'lodash/get';
+
+export default class EmailDomainUpgradeCtrl {
+  /* @ngInject */
+  constructor(
+    $translate,
+    $window,
+    Alerter,
+    EmailDomainService,
+    MANAGER_URLS,
+  ) {
+    this.$translate = $translate;
+    this.$window = $window;
+    this.Alerter = Alerter;
+    this.EmailDomainService = EmailDomainService;
+    this.MANAGER_URLS = MANAGER_URLS;
+  }
+
+  $onInit() {
+    this.alerts = {
+      main: 'webEmailDomainUpgrade.alerts.main',
+    };
+
+    this.model = {
+      offer: null,
+      duration: null,
+      agree: false,
+    };
+
+    this.loading = {
+      upgrading: false,
+      durations: false,
+    };
+
+    this.durations = [];
+  }
+
+  getDurations() {
+    this.loading.durations = true;
+    this.Alerter.resetMessage(this.alerts.main);
+
+    return this.EmailDomainService.getDuration(this.serviceName, this.model.offer.name)
+      .then(durations => {
+        this.durations = durations;
+        this.model.duration = head(durations);
+      })
+      .catch(() => {
+        this.durations = [];
+        this.Alerter.error(
+          this.$translate.instant('email_domain_upgrade_fetch_duration_error', {
+            message: get(error, 'data.message', error.message),
+          }),
+          this.alerts.main,
+        );
+      })
+      .finally(() => { this.loading.durations = false; });
+  }
+
+  upgrade() {
+    this.loading.upgrading = true;
+    this.Alerter.resetMessage(this.alerts.main);
+    return this.EmailDomainService.upgrade(this.serviceName, this.model.offer.name, this.model.duration.name)
+      .then(order => {
+        this.goToEmailDomain(
+          this.serviceName,
+          this.$translate.instant(
+            'email_domain_upgrade_success',
+            { orderTrackURL: `${this.MANAGER_URLS.dedicated}billing/order/${order.orderId}` },
+          ),
+        );
+        this.$window.open(order.url);
```

Comment:
I've got a suggestion for this 😉

Reply:
replaced it with this.$window.open(order.url, '_blank', 'noopener');